### PR TITLE
Global resource resize changes

### PR DESCRIPTION
**Pull request by abhinandanpb**
Changes to : 
1) Allows global settings like vlan and vxlan ranges
2) Unit tests
3) lock for global config modifications
4) Minor fixes

### CODE REVIEW

File: netmaster/gstate/gstate.go [hunk 1]
```diff
@@ -436,3 +488,40 @@ func (gc *Cfg) UnassignNetwork(networkName string) error {
...
+			maxUsed, _ = strconv.Atoi(strings.TrimSpace(tags[len(tags)-1]))
+		}
+	}
+
+	r, err := netutils.ParseTagRanges(ranges, pktTagType)
+	if err != nil {
+		return false
+	}
+
+	if r[0].Min > minUsed || r[0].Max < maxUsed {
```
Comment (jojimt) [inline netmaster/gstate/gstate.go]: 1) Do we allow disjoint  subranges within ranges? If so, should we be looking at r[len(r)-1].Max?
2) Should minUsed and maxUsed be swapped in the above expression to ensure ranges falls outside the used range?

File: netmaster/master/netmaster.go [hunk 2]
```diff
@@ -159,11 +162,105 @@ func CreateGlobal(stateDriver core.StateDriver, gc *intent.ConfigGlobal) error {
...
+	// check for valid values
+	if gc.NwInfraType != "" {
+		switch gc.NwInfraType {
+		case "default", "aci", "aci-opflex":
+			// These values are acceptable.
+		default:
+			return errors.New("Invalid fabric mode")
+		}
+		masterGc.NwInfraType = gc.NwInfraType
+	}
```
Comment (jojimt) [inline netmaster/master/netmaster.go]: Would be good to do this validation before we take the lock.

File: netmaster/objApi/objapi_test.go [hunk 3]
```diff
@@ -1081,6 +1082,124 @@ func TestNetworkAddDelete(t *testing.T) {
...
+	checkCreateNetwork(t, true, "default", "contiv5", "", "vlan", "14.1.1.1/24", "14.1.1.254", 302, "", "")
+
+	//verify creation of network succeeds after vlan change
+	checkCreateNetwork(t, false, "default", "contiv6", "", "vlan", "15.1.1.1/24", "15.1.1.254", 299, "", "")
+	checkInspectNetwork(t, false, "default", "contiv6", "15.1.1.254", 299, 0)
+	//checkInspectGlobal(t, false, "4", "")
+	verifyNetworkState(t, "default", "contiv6", "data", "vlan", "15.1.1.1", "15.1.1.254", 24, 299, 0, "", "", 0)
+
+	//check global state change fails if there are active tags
+	checkGlobalSet(t, true, "default", "50-301", "1-10000", "bridge")
```
Comment (jojimt) [inline netmaster/objApi/objapi_test.go]: we should also check the auto vlan allocation case.

File: netmaster/gstate/gstate.go (hunk 1 above)
Comment (abhinandanpb) [inline netmaster/gstate/gstate.go]: We dont support disjoint ranges for vxlan at the moment.  Unless there is a need due the amount of vxlan range space available. We can enhance it should need be

File: netmaster/master/netmaster.go (hunk 2 above)
Comment (abhinandanpb) [inline netmaster/master/netmaster.go]: sure we can do that. But we are anyways exiting if invalid? But I can make the change

File: netmaster/objApi/objapi_test.go (hunk 3 above)
Comment (abhinandanpb) [inline netmaster/objApi/objapi_test.go]: Sure I can add them

**Pull request merged**